### PR TITLE
Followup on pr 490

### PR DESCRIPTION
**Pull request by Ferenc-**
Fixes comments in https://github.com/zeux/pugixml/pull/490

### CODE REVIEW

File: src/pugixml.cpp [hunk 1]
```diff
@@ -5329,11 +5329,16 @@ namespace pugi
...
 	{
 		if (!_attr) return false;
 
-		return impl::strcpy_insitu(_attr->value, _attr->header, impl::xml_memory_page_value_allocated_mask, rhs, impl::strlength(rhs));
+		return impl::strcpy_insitu(_attr->value, _attr->header, impl::xml_memory_page_value_allocated_mask, rhs, sz);
+	}
+
+	PUGI__FN bool xml_attribute::set_value(const char_t* rhs)
+	{
+		return set_value(rhs, impl::strlength(rhs));
```
Comment (zeux) [inline src/pugixml.cpp:5341]: Should this have the original code as is? Alternatively, maybe other set_value methods should be calling the new overload with size? The asymmetry is odd.

**Comment by zeux** [comment]
Would you mind also adding tests for this to test_dom_traverse.cpp (for xml_node / xml_attribute) and test_dom_text.cpp (for xml_text)?

File: src/pugixml.cpp (hunk 1 above)
Comment (Ferenc-) [reply src/pugixml.cpp:5339]: You are absolutely right.This was never intended... Now I have changed it.

**Comment by Ferenc-** [comment]
> Would you mind also adding tests for this to test_dom_traverse.cpp (for xml_node / xml_attribute) and test_dom_text.cpp (for xml_text)?

I have addedd one to `test_dom_text.cpp` for the xml_text. If you find that TC acceptebale, then I will continue with  `test_dom_traverse.cpp`.

**Comment by zeux** [comment]
> If you find that TC acceptebale, then I will continue with test_dom_traverse.cpp.

Yup something like this is enough, just want to make sure this code is exercised at all by tests.

**Comment by Ferenc-** [comment]
> Yup something like this is enough, just want to make sure this code is exercised at all by tests.

Thanks for the quick feedback!  
I have added one more to `test_dom_text.cpp` and the rest into `test_dom_modify.cpp`  
since I figured that they fit there better, as these are modify functions.

File: src/pugixml.cpp (hunk 1 above)
Comment (zeux) [reply src/pugixml.cpp:5341]: Are you sure this was addressed?

**Comment by zeux** [comment]
> and the rest into test_dom_modify.cpp

Yes, thank you, `test_dom_modify.cpp` is indeed a better place.

**Comment by Ferenc-** [comment]
> > and the rest into test_dom_modify.cpp
> 
> Yes, thank you, `test_dom_modify.cpp` is indeed a better place.

Thanks for your feedback.
Is there anything else I can do to move this forward?

**Comment by Ferenc-** [comment]
Thank you so much for your effort in reviewing, extending and including this.  
Besides cheering me up personally, I believe it opens up new possibilities,
for example the packaging of one particular mapping software into various Linux distros.

**Pull request merged**